ethers-contract/src/call.rs

```diff
@@ -73,33 +72,45 @@ pub enum ContractError<M: Middleware> {
     ContractNotDeployed,
 }
 
+/// type alias for backwards compatibility
+pub type ContractCall<M, D> = ContractCallInternal<std::sync::Arc<M>, M, D>;
+
 #[derive(Debug)]
 #[must_use = "contract calls do nothing unless you `send` or `call` them"]
 /// Helper for managing a transaction before submitting it to a node
-pub struct ContractCall<M, D> {
+pub struct ContractCallInternal<B, M, D> {
```

Comment:
What is the motivation for this renaming?
so that we can use `ContractCall` for backwards compat?

The "internal" postfix somewhat irritates me as well.
I guess, technically this is a `ContractFunctionCall` or `FunctionCall`?

Reply:
`FunctionCall` seems like a good name, as contract calls may not have functions (e.g. fallback or receive invocations)